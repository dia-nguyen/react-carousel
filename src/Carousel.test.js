import { render, fireEvent } from "@testing-library/react";
import Carousel from "./Carousel";
import TEST_IMAGES from "./_testCommon.js";

/** Click right arrow n times */

function moveForward(container, n) {
  const rightArrow = container.querySelector('.bi-arrow-right-circle');
  for (let i = 0; i < n; i++) {
    fireEvent.click(rightArrow);
  }
}

/** Click left arrow n times */

function moveBackward(container, n) {
  const leftArrow = container.querySelector('.bi-arrow-left-circle');
  for (let i = 0; i < n; i++) {
    fireEvent.click(leftArrow);
  }
}

it("renders without crashing", function () {
  render(<Carousel photos={TEST_IMAGES} title="images for testing" />);
});

it("it matches snapshot", function () {
  const { container } = render(
    <Carousel photos={TEST_IMAGES} title="images for testing" />
  );

  expect(container).toMatchSnapshot();
});

it("works when you click on the right arrow", function () {
  const { container } = render(
    <Carousel photos={TEST_IMAGES} title="images for testing" />
  );
  // expect the first image to show, but not the second
  expect(
    container.querySelector('img[alt="testing image 1"]')
  ).toBeInTheDocument();
  expect(
    container.querySelector('img[alt="testing image 2"]')
  ).not.toBeInTheDocument();

  // move forward in the carousel
  moveForward(container, 1);

  // expect the second image to show, but not the first
  expect(
    container.querySelector('img[alt="testing image 1"]')
  ).not.toBeInTheDocument();
  expect(
    container.querySelector('img[alt="testing image 2"]')
  ).toBeInTheDocument();

  expect(container).toMatchSnapshot();
});

it("works when you click on the left arrow", function () {
  const { container } = render(
    <Carousel photos={TEST_IMAGES} title="images for testing" />
  );
  // move forward in the carousel
  moveForward(container, 1);

  // expect the second image to show, but not the first
  expect(
    container.querySelector('img[alt="testing image 1"]')
  ).not.toBeInTheDocument();
  expect(
    container.querySelector('img[alt="testing image 2"]')
  ).toBeInTheDocument();

  // move backward in the carousel
  moveBackward(container, 1);

  // expect the first image to show, but not the second
  expect(
    container.querySelector('img[alt="testing image 1"]')
  ).toBeInTheDocument();
  expect(
    container.querySelector('img[alt="testing image 2"]')
  ).not.toBeInTheDocument();

  expect(container).toMatchSnapshot();
});

it("hides the left arrow if on first image", function () {
  const { container } = render(
    <Carousel photos={TEST_IMAGES} title="images for testing" />
  );
  const leftArrow = container.querySelector(".bi-arrow-left-circle");
  const rightArrow = container.querySelector(".bi-arrow-right-circle");

  expect(leftArrow).not.toBeInTheDocument();
  // expect(rightArrow).toBeInTheDocument(); // Requires some fixing with the visibility
});

it("hides the right arrow if on last image", function () {
  const { container } = render(
    <Carousel photos={TEST_IMAGES} title="images for testing" />
  );

  // move forward in the carousel
  moveForward(container, 2)

  const leftArrow = container.querySelector(".bi-arrow-left-circle");
  const rightArrow = container.querySelector(".bi-arrow-right-circle");

  expect(rightArrow).not.toBeInTheDocument();
  // expect(leftArrow).toBeInTheDocument();
});