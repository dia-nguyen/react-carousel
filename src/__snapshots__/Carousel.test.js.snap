// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it matches snapshot 1`] = `
<div>
  <div
    class="Carousel"
  >
    <h1>
      images for testing
    </h1>
    <div
      class="Carousel-main"
    >
      <i
        class="bi bi-arrow-left-circle"
      />
      <div
        class="Card"
      >
        <h4
          class="Card-title"
        >
          testing image 1
        </h4>
        <img
          alt="testing image 1"
          class="Card-image"
          src="test1.com"
        />
        <small
          class="Card-small"
        >
          Image 
          1
           of 
          3
          .
        </small>
      </div>
      <i
        class="bi bi-arrow-right-circle"
      />
    </div>
  </div>
</div>
`;

exports[`matches next slide click 1`] = `
<div>
  <div
    class="Carousel"
  >
    <h1>
      images for testing
    </h1>
    <div
      class="Carousel-main"
    >
      <i
        class="bi bi-arrow-left-circle"
      />
      <div
        class="Card"
      >
        <h4
          class="Card-title"
        >
          testing image 2
        </h4>
        <img
          alt="testing image 2"
          class="Card-image"
          src="test2.com"
        />
        <small
          class="Card-small"
        >
          Image 
          2
           of 
          3
          .
        </small>
      </div>
      <i
        class="bi bi-arrow-right-circle"
      />
    </div>
  </div>
</div>
`;

exports[`matches prev slide click 1`] = `
<div>
  <div
    class="Carousel"
  >
    <h1>
      images for testing
    </h1>
    <div
      class="Carousel-main"
    >
      <i
        class="bi bi-arrow-left-circle"
      />
      <div
        class="Card"
      >
        <h4
          class="Card-title"
        >
          testing image 2
        </h4>
        <img
          alt="testing image 2"
          class="Card-image"
          src="test2.com"
        />
        <small
          class="Card-small"
        >
          Image 
          2
           of 
          3
          .
        </small>
      </div>
      <i
        class="bi bi-arrow-right-circle"
      />
    </div>
  </div>
</div>
`;

exports[`snapshot matches next slide click 1`] = `
<div>
  <div
    class="Carousel"
  >
    <h1>
      images for testing
    </h1>
    <div
      class="Carousel-main"
    >
      <i
        class="bi bi-arrow-left-circle"
      />
      <div
        class="Card"
      >
        <h4
          class="Card-title"
        >
          testing image 2
        </h4>
        <img
          alt="testing image 2"
          class="Card-image"
          src="test2.com"
        />
        <small
          class="Card-small"
        >
          Image 
          2
           of 
          3
          .
        </small>
      </div>
      <i
        class="bi bi-arrow-right-circle"
      />
    </div>
  </div>
</div>
`;

exports[`snapshot matches prev slide click 1`] = `
<div>
  <div
    class="Carousel"
  >
    <h1>
      images for testing
    </h1>
    <div
      class="Carousel-main"
    >
      <i
        class="bi bi-arrow-left-circle"
      />
      <div
        class="Card"
      >
        <h4
          class="Card-title"
        >
          testing image 2
        </h4>
        <img
          alt="testing image 2"
          class="Card-image"
          src="test2.com"
        />
        <small
          class="Card-small"
        >
          Image 
          2
           of 
          3
          .
        </small>
      </div>
      <i
        class="bi bi-arrow-right-circle"
      />
    </div>
  </div>
</div>
`;

exports[`works when you click on the left arrow 1`] = `
<div>
  <div
    class="Carousel"
  >
    <h1>
      images for testing
    </h1>
    <div
      class="Carousel-main"
    >
      <i
        class="bi bi-arrow-left-circle"
      />
      <div
        class="Card"
      >
        <h4
          class="Card-title"
        >
          testing image 1
        </h4>
        <img
          alt="testing image 1"
          class="Card-image"
          src="test1.com"
        />
        <small
          class="Card-small"
        >
          Image 
          1
           of 
          3
          .
        </small>
      </div>
      <i
        class="bi bi-arrow-right-circle"
      />
    </div>
  </div>
</div>
`;

exports[`works when you click on the right arrow 1`] = `
<div>
  <div
    class="Carousel"
  >
    <h1>
      images for testing
    </h1>
    <div
      class="Carousel-main"
    >
      <i
        class="bi bi-arrow-left-circle"
      />
      <div
        class="Card"
      >
        <h4
          class="Card-title"
        >
          testing image 2
        </h4>
        <img
          alt="testing image 2"
          class="Card-image"
          src="test2.com"
        />
        <small
          class="Card-small"
        >
          Image 
          2
           of 
          3
          .
        </small>
      </div>
      <i
        class="bi bi-arrow-right-circle"
      />
    </div>
  </div>
</div>
`;
